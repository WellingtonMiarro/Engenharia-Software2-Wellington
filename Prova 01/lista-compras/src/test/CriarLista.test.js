import React from 'react';

import CriarLista from '../pages/lista/CriarLista';

import toJson from 'enzyme-to-json';
//ele torna o snapshot ja em formato de texto
import {createSerializer} from 'enzyme-to-json';


// Utilizando para configuração do ENZYME
import { shallow, configure} from 'enzyme';
import  Adapter from 'enzyme-adapter-react-16';


configure({adapter: new Adapter()});
 expect.addSnapshotSerializer(createSerializer({mode:'deep'}));


describe("Teste de renderização do camponente CRIAR LISTA",()=>{
    it("Verificar se a página Inicial e carregada adequadamente", ()=>{
        const wrapper = shallow(<CriarLista/>);//retorna Serializado
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});