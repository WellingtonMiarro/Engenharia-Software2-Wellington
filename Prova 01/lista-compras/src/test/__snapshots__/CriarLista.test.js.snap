// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de renderização do camponente CRIAR LISTA Verificar se a página Inicial e carregada adequadamente 1`] = `
<div>
  <Menu
    logo="paginaAnterior.png"
    paginaAnterior="/"
    titulo="Criar lista"
  />
  <form
    id="formNovaLista"
    onSubmit={[Function]}
  >
    <input
      name="nome"
      onChange={[Function]}
      placeholder="Digite o nome da lista"
      required={true}
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Criar"
    />
  </form>
</div>
`;
